### PR TITLE
Use Cosmiconfig for configuration.

### PR DESCRIPTION
**Pull request by rkrisztian**
- Now all command-line options can be specified as configuration options in `package.json` or in other places that Cosmiconfig
  supports. In addition, configuration profiles can be defined that override such options. These features will help alleviate
  the problem of too long lines of NPM scripts using `gocd-cypress` in `package.json`.
- Some underscore-case configuration keys have been normalized to snake-case keys. (Breaks compatibility.)

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -84,4 +84,7 @@ const runWithStartCommand = async (serveCmd: string, serveHost: string, runCypre
 	}
 };
 
-export const runHandler = dockerize(runCommand);
+export const runHandler = (argv: Arguments<ArgTypes>) => {
+	loadConfig(argv);
+	dockerize(runCommand)();
```
Comment (rkrisztian) [inline src/runner.ts]: Now we kind of break the original functional style which used function composition, and that's a bit counter-intuitive now...

Comment (rkrisztian) [reply src/runner.ts]: See my new commit, which addresses the problem. I've removed the use of the function composition technique, as I didn't find it beneficial for this case. The `runCommand` is simply not called (in the same script invocation) if we use Docker.

**Comment by rkrisztian** [comment]
One thing that's worth to know but forgot to add to the commit message is, I've considered yargs' `.config(...)` builder, but it failed for two reasons: 

1. Either it wants a JSON file or an object literal. Object literal isn't okay for the 2nd point, and just JSON is more restrictive than what Cosmiconfig allows.
2. I needed to support profiles which could easily run into a chicken-and-egg problem.

**Review by nemeth-z — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -39,42 +39,74 @@ npx gocd-cypress --help
...
+
+Example in `package.json`:
+```json
+{
+    "name": "my-project",
+    "version": "0.0.0",
+    "gocdCypress": {
+        "cypressCmd": "cypress run --browser firefox",
+        "dockerImage": "cypress/browsers:node16.13.0-chrome95-ff94"
+    }
```
Comment (nemeth-z) [inline README.md:53]: Nice! This and the below config option looks much better than the previous solution. 👍

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
 	"license": "MIT",
 	"dependencies": {
 		"chalk": "^4.1.2",
+		"cosmiconfig": "^7.0.1",
```
Comment (hosszubalazs) [inline package.json:21]: Lets hope the tool finds new owners and does not get abandoned.

**Review by hosszubalazs — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (rkrisztian) [reply package.json:21]: I agree, it's a bit of a risky move, but it's such a useful tool with lots of GitHub stars (quite popular), so IMO it has pretty good chances to survive.

**Comment by rkrisztian** [comment]
Unfortunately my tests did not point out a bug that yargs defaults can override cosmiconfig values. I'll fix it in the next PR.

**Edit:** See https://github.com/adnovum/gocd-cypress/pull/13

**Pull request merged**